types/chrome-remote-interface/chrome-remote-interface-tests.ts

```diff
@@ -41,6 +46,13 @@ import CDP = require('chrome-remote-interface');
                 await CDP.Close({ ...cdpPort, id: target.id });
             }
         }
+
+        assertType<Promise<void>>(client.send('Network.enable'));
+        assertType<Promise<Protocol.Page.NavigateResponse>>(client.send('Page.navigate', {url: 'https://github.com'}));
+        assertType<Promise<Protocol.Page.NavigateResponse>>(client.send('Page.navigate', {url: 'https://github.com'}, 'sessionId'));
+        client.send('Page.navigate', (error: boolean, response: Protocol.Page.NavigateResponse | CDP.SendError) => {});
+        client.send('Page.navigate', {url: 'https://github.com'}, (error: boolean, response: Protocol.Page.NavigateResponse | CDP.SendError) => {});
+        client.send('Page.navigate', {url: 'https://github.com'}, 'sessionId', (error: boolean, response: Protocol.Page.NavigateResponse | CDP.SendError) => {});
```

Comment:
```suggestion
        client.send('Page.navigate', {url: 'https://github.com'}, 'sessionId', (error: boolean, response: Protocol.Page.NavigateResponse | CDP.SendError) => {});
        // @ts-expect-error
        client.send('Page.navigate', (error: boolean, response: Protocol.Page.NavigateResponse) => {});
```

The last line above should be erroneous, but apparently (I have no idea why) if one wants TypeScript to indicate that, one has to define the overloaded callback as follows instead of using `((...) => ...) | ((...) => ...)`:

```ts
interface SendCallback<T extends keyof ProtocolMappingApi.Commands> {
    (error: false, response: ProtocolMappingApi.Commands[T]['returnType']): void;
    (error: true, response: SendError): void;
}
```

thus the relevant `send()` definitions become:

```ts
send<T extends keyof ProtocolMappingApi.Commands>(event: T, callback: SendCallback<T>): void;
send<T extends keyof ProtocolMappingApi.Commands>(event: T, params: ProtocolMappingApi.Commands[T]['paramsType'][0],
    callback: SendCallback<T>): void;
send<T extends keyof ProtocolMappingApi.Commands>(event: T, params: ProtocolMappingApi.Commands[T]['paramsType'][0], sessionId: string,
    callback: SendCallback<T>): void;
```

Btw, `npm test chrome-remote-interface` handles `@ts-expect-error` just fine but it was introduced in [TypeScript 3.9](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments). Unfortunately, VSCode doesn't recognize `$ExpectError`.

Reply:
@westy92 Is everything ok? Are there any problems?

Reply:
Ideally, this type narrowing would work too:
![image](https://user-images.githubusercontent.com/290167/129833661-4ea00e08-6c13-44de-abc4-ffef448d56c2.png)

I'm toying around with things. 😅

Reply:
I made the change. It's better, but not perfect. AKA good enough. :)
Thank you for your help!

Reply:
> Ideally, this type narrowing would work too:
> ![image](https://user-images.githubusercontent.com/290167/129833661-4ea00e08-6c13-44de-abc4-ffef448d56c2.png)

Yeah probably best to let the compiler infer parameter types :sweat_smile: ... Compiler recognition of the type narrowing should be eminently doable (I think) but a deep dive into the compiler doesn't seem particularly appealing at this moment :joy: